Commander/vault/SecretManagerCommands.cs

```diff
@@ -389,9 +409,247 @@ private static void DumpSecretManagerApplicationInfo(VaultData vault, SecretsMan
+                { "removed", removedUsers }
+            };
+
+            var ShareFolderRequests = new List<Task>();
+            var ShareRecordRequests = new List<Task>();
+
+            foreach (var appUser in appUsersMap)
+            {
+                var group = appUser.Key;
+                var usersList = appUser.Value;
+                if (usersList.Count == 0) continue;
+                var userTasks = usersList.Select(async x => new { User = x, NeedsUpdate = await UserNeedsUpdateAsync(vault, x, group == "admins", shareUids, applicationUid) }).ToList();
+                var userResults = await Task.WhenAll(userTasks);
+                var users = userResults.Where(r => r.NeedsUpdate).Select(r => r.User).ToList();
+
+                var shareFolderAction = removed != null ? "remove" : "grant";
+                var shareRecordAction = removed != null ? "revoke" : "share";
+
+                var sfUpdates = new List<(string FolderUid, string User)>();
+                await HandleSharedFolderShare(vault, applicationUid, sharedFolders, group, users, shareFolderAction);
+                await HandleRecordsShare(vault, applicationUid, recordShares, group, users, shareRecordAction);
+
+            }
+        }
+
+        private static async Task HandleRecordsShare(VaultOnline vault, string applicationUid, IEnumerable<RecordSharePermissions> recordShares, string group, List<string> users, string shareRecordAction)
+        {
+            foreach (var record in recordShares)
+            {
+                foreach (var user in users)
+                {
+                    if (await ShareNeedsUpdate(vault, user, record.RecordUid, group == "admins", applicationUid))
+                    {
+                        if (shareRecordAction == "revoke")
+                        {
+                            await vault.RevokeShareFromUser(record.RecordUid, user);
+                        }
+                        else
+                        {
+                            var sharedRecordOptions = new SharedFolderRecordOptions
+                            {
+                                CanEdit = group == "admins",
+                                CanShare = group == "admins"
+                            };
+                            await vault.ShareRecordWithUser(record.RecordUid, user, sharedRecordOptions);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static async Task HandleSharedFolderShare(VaultOnline vault, string applicationUid, List<string> sharedFolders, string group, List<string> users, string shareFolderAction)
+        {
+            foreach (var folder in sharedFolders)
+            {
+                foreach (var user in users)
+                {
+                    if (await ShareNeedsUpdate(vault, user, folder, group == "admins", applicationUid))
+                    {
+                        if (shareFolderAction == "remove")
```

Comment:
Share Action must of finite values, consider using enum instead of the hardcoded strings.